cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml

```diff
@@ -12,7 +12,7 @@ license = "Apache 2.0"
 [tool.poetry.dependencies]
 python = "<3.11,>=3.7.1"
 requests = "^2.25.1"
-singer-sdk = "^0.14.0"
+singer-sdk = { version="^0.14.0", extras = ["s3"] }
```

Comment:
This will install the extras in all derived packages. Passthrough extras are unfortunately not supported (see https://github.com/orgs/python-poetry/discussions/4313).

Can we make an explicit extra for the S3 dependencies?

For example:

```toml
[tool.poetry.dependencies]
python = "<3.11,>=3.7.1"
requests = "^2.25.1"
singer-sdk = { version="^0.14.0"}
fs-s3fs = { version = "^1.1.1", optional = true}

[tool.poetry.extras]
s3 = ["fs-s3fs"]
```

Reply:
Thanks Edgar. I've updated the pyproject.toml tap/target templates to include an explicit extra for s3 dependencies.